### PR TITLE
Add missing '&' in mask calculation

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**

### CODE REVIEW

**Comment by DimitriPapadopoulos** [comment]
Run tests:
```
$ nosetests ./tests
/volatile/src/pylearn-parsimony/tests/test_linear_regression_large.py:28: UserWarning: Download of weight map is not authorized and it is required to complete this test.
Set allow_downloads = True in the file: config.ini, if you want to run this test!
  warnings.warn("Download of weight map is not authorized and it is "
/volatile/src/pylearn-parsimony/tests/test_logistic_regression_large.py:32: UserWarning: Download of weight map is not authorized and it is required to complete this test.
Set allow_downloads = True in the file: config.ini, if you want to run this test!
  warnings.warn("Download of weight map is not authorized and it is "
............................................/usr/lib/python2.7/dist-packages/sklearn/linear_model/coordinate_descent.py:491: ConvergenceWarning: Objective did not converge. You might want to increase the number of iterations. Fitting data with very small alpha may cause precision problems.
  ConvergenceWarning)
............................................../usr/lib/python2.7/dist-packages/sklearn/linear_model/stochastic_gradient.py:128: FutureWarning: max_iter and tol parameters have been added in <class 'sklearn.linear_model.stochastic_gradient.SGDClassifier'> in 0.19. If both are left unset, they default to max_iter=5 and tol=None. If tol is not None, max_iter defaults to max_iter=1000. From 0.21, default max_iter will be 1000, and default tol will be 1e-3.
  "and default tol will be 1e-3." % type(self), FutureWarning)
......../volatile/src/pylearn-parsimony/parsimony/functions/combinedfunctions.py:1602: RuntimeWarning: divide by zero encountered in log
  l_star = np.sum((b * np.log(b) + (1 - b)
/volatile/src/pylearn-parsimony/parsimony/functions/combinedfunctions.py:1602: RuntimeWarning: invalid value encountered in multiply
  l_star = np.sum((b * np.log(b) + (1 - b)
/volatile/src/pylearn-parsimony/parsimony/functions/combinedfunctions.py:1603: RuntimeWarning: divide by zero encountered in log
  * np.log(1 - b)) * (self.weights * scale))
/volatile/src/pylearn-parsimony/parsimony/functions/combinedfunctions.py:1603: RuntimeWarning: invalid value encountered in multiply
  * np.log(1 - b)) * (self.weights * scale))
.............................../volatile/src/pylearn-parsimony/parsimony/algorithms/proximal.py:533: UserWarning: Stopping criterion satisfied before the first iteration. Either beta_start a the solution (given eps). If beta_start is null the problem might be over-penalized.  Then try smaller penalization.
  "Stopping criterion satisfied before the first iteration."
....
...................
----------------------------------------------------------------------
Ran 152 tests in 1445.670s

OK
```
